openedx/core/djangoapps/programs/utils.py

```diff
@@ -450,6 +451,78 @@ def _attach_course_run_upgrade_url(self, run_mode):
     def _attach_course_run_may_certify(self, run_mode):
         run_mode['may_certify'] = self.course_overview.may_certify()
 
```

Comment:
Note: Moved this in to to the parent class so that it could be used by both classes. This data is valuable in both cases.